src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseDataSourceServiceImpl.java

```diff
@@ -117,6 +120,18 @@ public BaseDataSourceServiceImpl(BaseSolrService<SF> solrService,
         this.studyDataSourceLinkRepository = studyDataSourceLinkRepository;
     }
 
+    protected void beforeCreate(final DS dataSource, final boolean virtual) {
+
+        dataSource.setPublished(false);
+        dataSource.setCreated(new Date());
+        final Set<Tenant> tenants = tenantService.getDefault();
```

Comment:
Is there really a need in default tenants for virtual source? Why not if-else then?